In [1]:
import scanpy as sc 
import pandas as pd

In [2]:
outpath="/home/lugli/spuccio/Projects/SP039/FileH5AD_input/"

In [3]:
df = pd.read_csv("/home/lugli/spuccio/Projects/SP039/GSE131928_PMID31327527/GSM3828673_10X_GBM_IDHwt_processed_TPM.tsv.gz",sep="\t")

In [4]:
df

GENE  102_1       102_2  102_4  102_5  102_7  102_8  \
0            FAM138A    0.0    0.000000    0.0    0.0    0.0    0.0   
1              OR4F5    0.0    0.000000    0.0    0.0    0.0    0.0   
2       RP11-34P13.7    0.0    0.000000    0.0    0.0    0.0    0.0   
3       RP11-34P13.8    0.0    0.000000    0.0    0.0    0.0    0.0   
4      RP11-34P13.14    0.0    0.000000    0.0    0.0    0.0    0.0   
...              ...    ...         ...    ...    ...    ...    ...   
30309     AL592183.1    0.0  157.678966    0.0    0.0    0.0    0.0   
30310     AC011841.1    0.0    0.000000    0.0    0.0    0.0    0.0   
30311     BX072566.1    0.0    0.000000    0.0    0.0    0.0    0.0   
30312     AL354822.1    0.0    0.000000    0.0    0.0    0.0    0.0   
30313          BAGE5    0.0    0.000000    0.0    0.0    0.0    0.0   

          102_10  102_11     102_12  ...  126_2_227  126_2_230  126_2_231  \
0        0.00000     0.0    0.00000  ...        0.0        0.0        0.0   
1        0.00000     0.0    0.00000  ...        0.0        0.0        0.0   
2        0.00000     0.0    0.00000  ...        0.0        0.0        0.0   
3        0.00000     0.0    0.00000  ...        0.0        0.0        0.0   
4        0.00000     0.0    0.00000  ...        0.0        0.0        0.0   
...          ...     ...        ...  ...        ...        ...        ...   
30309  246.00246     0.0  208.94275  ...        0.0        0.0        0.0   
30310    0.00000     0.0    0.00000  ...        0.0        0.0        0.0   
30311    0.00000     0.0    0.00000  ...        0.0        0.0        0.0   
30312    0.00000     0.0  208.94275  ...        0.0        0.0        0.0   
30313    0.00000     0.0    0.00000  ...        0.0        0.0        0.0   

       126_2_232  126_2_235  126_2_236  126_2_237  126_2_240  126_2_242  \
0            0.0        0.0        0.0        0.0        0.0        0.0   
1            0.0        0.0        0.0        0.0        0.0        0.0   
2            0.0        0.0        0.0        0.0        0.0        0.0   
3            0.0        0.0        0.0        0.0        0.0        0.0   
4            0.0        0.0        0.0        0.0        0.0        0.0   
...          ...        ...        ...        ...        ...        ...   
30309        0.0        0.0        0.0        0.0        0.0        0.0   
30310        0.0        0.0        0.0        0.0        0.0        0.0   
30311        0.0        0.0        0.0        0.0        0.0        0.0   
30312        0.0        0.0        0.0        0.0        0.0        0.0   
30313        0.0        0.0        0.0        0.0        0.0        0.0   

       126_2_245  
0            0.0  
1            0.0  
2            0.0  
3            0.0  
4            0.0  
...          ...  
30309        0.0  
30310        0.0  
30311        0.0  
30312        0.0  
30313        0.0  

[30314 rows x 16202 columns]

In [5]:
df.set_index(df["GENE"].str.split(".",expand=True)[0],inplace=True)

In [6]:
df = df.drop_duplicates(["GENE"])

In [7]:
df = df.T

In [8]:
df = df.iloc[1: , :]

df

0         FAM138A OR4F5 RP11-34P13 RP11-34P13 RP11-34P13 RP11-34P13  \
102_1         0.0   0.0        0.0        0.0        0.0        0.0   
102_2         0.0   0.0        0.0        0.0        0.0        0.0   
102_4         0.0   0.0        0.0        0.0        0.0        0.0   
102_5         0.0   0.0        0.0        0.0        0.0        0.0   
102_7         0.0   0.0        0.0        0.0        0.0        0.0   
...           ...   ...        ...        ...        ...        ...   
126_2_236     0.0   0.0        0.0        0.0        0.0        0.0   
126_2_237     0.0   0.0        0.0        0.0        0.0        0.0   
126_2_240     0.0   0.0        0.0        0.0        0.0        0.0   
126_2_242     0.0   0.0        0.0        0.0        0.0        0.0   
126_2_245     0.0   0.0        0.0        0.0        0.0        0.0   

0            AP006222 RP4-669L17 OR4F29 RP4-669L17  ...       MT-ND5  \
102_1             0.0        0.0    0.0        0.0  ...   691.204424   
102_2             0.0        0.0    0.0        0.0  ...   315.357931   
102_4             0.0        0.0    0.0        0.0  ...          0.0   
102_5             0.0        0.0    0.0        0.0  ...  2839.372634   
102_7             0.0        0.0    0.0        0.0  ...   804.667069   
...               ...        ...    ...        ...  ...          ...   
126_2_236         0.0        0.0    0.0        0.0  ...   393.287887   
126_2_237  216.450216        0.0    0.0        0.0  ...   649.350649   
126_2_240         0.0        0.0    0.0        0.0  ...   502.815768   
126_2_242         0.0        0.0    0.0        0.0  ...          0.0   
126_2_245         0.0        0.0    0.0        0.0  ...   253.613999   

0              MT-ND6        MT-CYB MGC39584 AC011043    AL592183 AC011841  \
102_1             0.0   8035.251426      0.0      0.0         0.0      0.0   
102_2      315.357931   3784.295175      0.0      0.0  157.678966      0.0   
102_4             0.0   2067.539628      0.0      0.0         0.0      0.0   
102_5             0.0  10816.657653      0.0      0.0         0.0      0.0   
102_7             0.0   5632.669483      0.0      0.0         0.0      0.0   
...               ...           ...      ...      ...         ...      ...   
126_2_236  131.095962    262.191924      0.0      0.0         0.0      0.0   
126_2_237         0.0   1948.051948      0.0      0.0         0.0      0.0   
126_2_240  201.126307    402.252615      0.0      0.0         0.0      0.0   
126_2_242         0.0           0.0      0.0      0.0         0.0      0.0   
126_2_245     126.807       126.807      0.0      0.0         0.0      0.0   

0         BX072566 AL354822 BAGE5  
102_1          0.0      0.0   0.0  
102_2          0.0      0.0   0.0  
102_4          0.0      0.0   0.0  
102_5          0.0      0.0   0.0  
102_7          0.0      0.0   0.0  
...            ...      ...   ...  
126_2_236      0.0      0.0   0.0  
126_2_237      0.0      0.0   0.0  
126_2_240      0.0      0.0   0.0  
126_2_242      0.0      0.0   0.0  
126_2_245      0.0      0.0   0.0  

[16201 rows x 30314 columns]

In [9]:
Metadata = pd.read_excel("/home/lugli/spuccio/Projects/SP039/GSE131928_PMID31327527/GSE131928_single_cells_tumor_name_and_adult_or_peidatric.xlsx",skiprows=[i for i in range(0,43)])

In [10]:
Metadata = Metadata.set_index("Sample name")

In [11]:
bigtable = pd.merge(df,Metadata,left_index=True,right_index=True) 

In [12]:
bigtable['tumour name'].unique() 

array(['MGH102', 'MGH143', 'MGH105', 'MGH114', 'MGH115', 'MGH118',
       'MGH124', 'MGH125', 'MGH126'], dtype=object)

In [13]:
bigtable = bigtable.loc[bigtable['tumour name'] != "MGH143"]

In [14]:
bigtable['tumour name'].unique() 

array(['MGH102', 'MGH105', 'MGH114', 'MGH115', 'MGH118', 'MGH124',
       'MGH125', 'MGH126'], dtype=object)

In [15]:
bigtable = bigtable.set_index('MGH' +bigtable.index.astype(str))

In [16]:
bigtable = bigtable.set_index(bigtable.index.astype(str).str.replace('_','-'))

In [17]:
MGH102 = bigtable.loc[bigtable['tumour name'] == "MGH102"]
MGH105 = bigtable.loc[bigtable['tumour name'] == "MGH105"]
MGH114 = bigtable.loc[bigtable['tumour name'] == "MGH114"]
MGH115 = bigtable.loc[bigtable['tumour name'] == "MGH115"]
MGH118 =  bigtable.loc[bigtable['tumour name'] == "MGH118"]
MGH124 =  bigtable.loc[bigtable['tumour name'] == "MGH124"]
MGH125 =  bigtable.loc[bigtable['tumour name'] == "MGH125"]
MGH126 =  bigtable.loc[bigtable['tumour name'] == "MGH126"]

In [18]:
MGH102 = MGH102.reset_index(drop=True)
MGH105 = MGH105.reset_index(drop=True)
MGH114 = MGH114.reset_index(drop=True)
MGH115 = MGH115.reset_index(drop=True)
MGH118 = MGH118.reset_index(drop=True)
MGH124 = MGH124.reset_index(drop=True)
MGH125 = MGH125.reset_index(drop=True)
MGH126 = MGH126.reset_index(drop=True)

In [19]:
MGH102.set_index("MGH102-" + MGH102.index.astype(str),inplace=True)
MGH105.set_index("MGH102-" + MGH105.index.astype(str),inplace=True)
MGH114.set_index("MGH114-" + MGH114.index.astype(str),inplace=True)
MGH115.set_index("MGH115-" + MGH115.index.astype(str),inplace=True)
MGH118.set_index("MGH118-" + MGH118.index.astype(str),inplace=True)
MGH124.set_index("MGH124-" + MGH124.index.astype(str),inplace=True)
MGH125.set_index("MGH125-" + MGH125.index.astype(str),inplace=True)
MGH126.set_index("MGH126-" + MGH126.index.astype(str),inplace=True)

In [20]:
adataMGH102 = sc.AnnData(MGH102.iloc[: , :30314])
adataMGH105 = sc.AnnData(MGH105.iloc[: , :30314])
adataMGH114 = sc.AnnData(MGH114.iloc[: , :30314])
adataMGH115 = sc.AnnData(MGH115.iloc[: , :30314])
adataMGH118 = sc.AnnData(MGH118.iloc[: , :30314])
adataMGH124 = sc.AnnData(MGH124.iloc[: , :30314])
adataMGH125 = sc.AnnData(MGH125.iloc[: , :30314])
adataMGH126 = sc.AnnData(MGH126.iloc[: , :30314])

/tmp/ipykernel_15766/238673402.py:1: FutureWarning: X.dtype being converted to np.float32 from object. In the next version of anndata (0.9) conversion will not be automatic. Pass dtype explicitly to avoid this warning. Pass `AnnData(X, dtype=X.dtype, ...)` to get the future behavour.
  adataMGH102 = sc.AnnData(MGH102.iloc[: , :30314])
/home/lugli/spuccio/anaconda3/envs/scanpy19/lib/python3.10/site-packages/anndata/_core/anndata.py:1830: UserWarning: Variable names are not unique. To make them unique, call `.var_names_make_unique`.
  utils.warn_names_duplicates("var")
/tmp/ipykernel_15766/238673402.py:2: FutureWarning: X.dtype being converted to np.float32 from object. In the next version of anndata (0.9) conversion will not be automatic. Pass dtype explicitly to avoid this warning. Pass `AnnData(X, dtype=X.dtype, ...)` to get the future behavour.
  adataMGH105 = sc.AnnData(MGH105.iloc[: , :30314])
/tmp/ipykernel_15766/238673402.py:3: FutureWarning: X.dtype being converted to np.float32

In [21]:
adataMGH102.shape

(1822, 30314)

In [22]:
adataMGH105.shape

(5513, 30314)

In [23]:
adataMGH114.shape

(473, 30314)

In [24]:
adataMGH115.shape

(1283, 30314)

In [25]:
adataMGH118.shape

(539, 30314)

In [26]:
adataMGH124.shape

(2415, 30314)

In [27]:
adataMGH125.shape

(1613, 30314)

In [28]:
adataMGH126.shape

(229, 30314)

In [29]:
adataMGH102.obs['SampleID'] = "MGH102"
adataMGH102.obs['GEO_RNA'] = "NotAvailable"
adataMGH102.obs['Cohort'] = "GSE131928"
adataMGH102.obs['Type'] = "PrimaryTumor"  
adataMGH102.obs['Subtype'] = "NotAvailable"
adataMGH102.obs['Diagnosis'] = "Glioblastoma"
adataMGH102.obs['Grade'] = "IV"
adataMGH102.obs['Location'] = "Left_Temporal"
adataMGH102.obs['IDH1_Status'] = "WT"
adataMGH102.obs['EGFR_Status'] = "WT"
adataMGH124.obs['MET'] = "WT"
adataMGH102.obs['MGMT'] = "NM"
adataMGH102.obs['CellCount'] = "1822"
adataMGH102.obs['Sample'] = "Patient7"
adataMGH102.obs['Sex'] = "M"
adataMGH102.obs['Age'] = "65"
adataMGH102.obs['Sorted'] = "CD45"

In [30]:
adataMGH105.obs['SampleID'] = "MGH105"
adataMGH105.obs['GEO_RNA'] = "NotAvailable"
adataMGH105.obs['Cohort'] = "GSE131928"
adataMGH105.obs['Type'] = "PrimaryTumor"  
adataMGH105.obs['Subtype'] = "NotAvailable"
adataMGH105.obs['Diagnosis'] = "Glioblastoma"
adataMGH105.obs['Grade'] = "IV"
adataMGH105.obs['Location'] = "Right_Temporal"
adataMGH105.obs['IDH1_Status'] = "WT"
adataMGH105.obs['EGFR_Status'] = "Positive"
adataMGH105.obs['MET'] = "WT"
adataMGH105.obs['MGMT'] = "NM"
adataMGH105.obs['CellCount'] = "5513"
adataMGH105.obs['Sample'] = "Patient8"
adataMGH105.obs['Sex'] = "F"
adataMGH105.obs['Age'] = "67"
adataMGH105.obs['Sorted'] = "CD45"

In [31]:
adataMGH115.obs['SampleID'] = "MGH115"
adataMGH115.obs['GEO_RNA'] = "NotAvailable"
adataMGH115.obs['Cohort'] = "GSE131928"
adataMGH115.obs['Type'] = "PrimaryTumor"  
adataMGH115.obs['Subtype'] = "NotAvailable"
adataMGH115.obs['Diagnosis'] = "Glioblastoma"
adataMGH115.obs['Grade'] = "IV"
adataMGH115.obs['Location'] = "Right_Frontal"
adataMGH115.obs['IDH1_Status'] = "WT"
adataMGH115.obs['EGFR_Status'] = "Positive"
adataMGH124.obs['MET'] = "WT"
adataMGH115.obs['MGMT'] = "NM"
adataMGH115.obs['CellCount'] = "1283"
adataMGH115.obs['Sample'] = "Patient9"
adataMGH115.obs['Sex'] = "F"
adataMGH115.obs['Age'] = "69"
adataMGH115.obs['Sorted'] = "CD45"

In [32]:
adataMGH124.obs['SampleID'] = "MGH124"
adataMGH124.obs['GEO_RNA'] = "NotAvailable"
adataMGH124.obs['Cohort'] = "GSE131928"
adataMGH124.obs['Type'] = "PrimaryTumor"  
adataMGH124.obs['Subtype'] = "NotAvailable"
adataMGH124.obs['Diagnosis'] = "Glioblastoma"
adataMGH124.obs['Grade'] = "IV"
adataMGH124.obs['Location'] = "Right_Frontal"
adataMGH124.obs['IDH1_Status'] = "WT"
adataMGH124.obs['EGFR_Status'] = "WT"
adataMGH124.obs['MET'] = "WT"
adataMGH124.obs['MGMT'] = "NM"
adataMGH124.obs['CellCount'] = "2415"
adataMGH124.obs['Sample'] = "Patient10"
adataMGH124.obs['Sex'] = "F"
adataMGH124.obs['Age'] = "68"
adataMGH124.obs['Sorted'] = "CD45"

In [33]:
adataMGH125.obs['SampleID'] = "MGH125"
adataMGH125.obs['GEO_RNA'] = "NotAvailable"
adataMGH125.obs['Cohort'] = "GSE131928"
adataMGH125.obs['Type'] = "PrimaryTumor"  
adataMGH125.obs['Subtype'] = "NotAvailable"
adataMGH125.obs['Diagnosis'] = "Glioblastoma"
adataMGH125.obs['Grade'] = "IV"
adataMGH125.obs['Location'] = "Right_Frontatemporal"
adataMGH125.obs['IDH1_Status'] = "WT"
adataMGH125.obs['EGFR_Status'] = "WT"
adataMGH125.obs['MET'] = "WT"
adataMGH125.obs['MGMT'] = "NM"
adataMGH125.obs['CellCount'] = "1613"
adataMGH125.obs['Sample'] = "Patient11"
adataMGH125.obs['Sex'] = "M"
adataMGH125.obs['Age'] = "76"
adataMGH125.obs['Sorted'] = "CD45"

In [34]:
adataMGH126.obs['SampleID'] = "MGH126"
adataMGH126.obs['GEO_RNA'] = "NotAvailable"
adataMGH126.obs['Cohort'] = "GSE131928"
adataMGH126.obs['Type'] = "PrimaryTumor"  
adataMGH126.obs['Subtype'] = "NotAvailable"
adataMGH126.obs['Diagnosis'] = "Glioblastoma"
adataMGH126.obs['Grade'] = "IV"
adataMGH126.obs['Location'] = "Right_Temporal"
adataMGH126.obs['IDH1_Status'] = "WT"
adataMGH126.obs['EGFR_Status'] = "WT"
adataMGH126.obs['MET'] = "WT"
adataMGH126.obs['MGMT'] = "Methylated"
adataMGH126.obs['CellCount'] = "229"
adataMGH126.obs['Sample'] = "Patient12"
adataMGH126.obs['Sex'] = "M"
adataMGH126.obs['Age'] = "65"
adataMGH126.obs['Sorted'] = "CD45"

adataMGH114.obs['SampleID'] = "MGH114"
adataMGH114.obs['GEO_RNA'] = "NotAvailable"
adataMGH114.obs['Cohort'] = "GSE131928"
adataMGH114.obs['Type'] = "PrimaryTumor"  
adataMGH114.obs['Subtype'] = "NotAvailable"
adataMGH114.obs['Diagnosis'] = "Glioblastoma"
adataMGH114.obs['Grade'] = "NotAvailable"
adataMGH114.obs['Location'] = "NotAvailable"
adataMGH114.obs['IDH1_Status'] = "NotAvailable"
adataMGH114.obs['EGFR_Status'] = "NotAvailable"
adataMGH114.obs['MET'] = "NotAvailable"
adataMGH114.obs['MGMT'] = "NotAvailable"
adataMGH114.obs['CellCount'] = "473"
adataMGH114.obs['Sample'] = "Patient13"
adataMGH114.obs['Sex'] = "NotAvailable"
adataMGH114.obs['Age'] = "NotAvailable"
adataMGH114.obs['Sorted'] = "CD45"

adataMGH118.obs['SampleID'] = "MGH118"
adataMGH118.obs['GEO_RNA'] = "NotAvailable"
adataMGH118.obs['Cohort'] = "GSE131928"
adataMGH118.obs['Type'] = "PrimaryTumor"  
adataMGH118.obs['Subtype'] = "NotAvailable"
adataMGH118.obs['Diagnosis'] = "Glioblastoma"
adataMGH118.obs['Grade'] = "NotAvailable"
adataMGH118.obs['Location'] = "NotAvailable"
adataMGH118.obs['IDH1_Status'] = "NotAvailable"
adataMGH118.obs['EGFR_Status'] = "NotAvailable"
adataMGH118.obs['MET'] = "NotAvailable"
adataMGH118.obs['MGMT'] = "NotAvailable"
adataMGH118.obs['CellCount'] = "539"
adataMGH118.obs['Sample'] = "Patient14"
adataMGH118.obs['Sex'] = "NotAvailable"
adataMGH118.obs['Age'] = "NotAvailable"
adataMGH118.obs['Sorted'] = "CD45"

In [35]:
adataMGH102.var_names_make_unique()
adataMGH114.var_names_make_unique()
adataMGH115.var_names_make_unique()
adataMGH118.var_names_make_unique()
adataMGH124.var_names_make_unique()
adataMGH125.var_names_make_unique()
adataMGH126.var_names_make_unique()

In [36]:
adataMGH102.write(outpath+"MGH102.h5ad")
adataMGH105.write(outpath+"MGH105.h5ad")
#adataMGH114.write(outpath+"MGH114.h5ad")
adataMGH115.write(outpath+"MGH115.h5ad")
#adataMGH118.write(outpath+"MGH118.h5ad")
adataMGH124.write(outpath+"MGH124.h5ad")
adataMGH125.write(outpath+"MGH125.h5ad")
adataMGH126.write(outpath+"MGH126.h5ad")